### PR TITLE
192 Fade in / out the Clock automatically 

### PR DESCRIPTION
**Pull request by mastermichi**
- initial version

### CODE REVIEW

File: res/values-de/strings.xml [hunk 1]
```diff
@@ -266,7 +266,9 @@ Please preserve the xml tags and the utf8 encoding.
     <string name="remaining">übrig</string>
     <string name="repeat">wiederholen</string>
     <string name="reset_to_defaults">Einstellungen zurücksetzen</string>
+    <string name="pburnin">Einbrennen der Uhr verhindern</string>
```
Comment (firebirdberlin) [inline res/values-de/strings.xml:269]: Please don't abbreviate. Ideally the variable name explains in which context the string is used. I prefer camelCase notation over snail case.

File: res/values/prevent_burn_in.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string-array name="pburninModeValues">
```
Comment (firebirdberlin) [inline res/values/prevent_burn_in.xml:3]: Do you have a more speaking suggestion for the name ?

File: src/com/firebirdberlin/nightdream/ui/NightDreamUI.java [hunk 3]
```diff
@@ -519,7 +519,7 @@ public boolean onLongClick(View v) {
 
     public void onStart() {
         Log.d(TAG, "onStart()");
-        handler.postDelayed(moveAround, 30000);
+        handler.postDelayed(moveAround, 60000);
```
Comment (firebirdberlin) [inline src/com/firebirdberlin/nightdream/ui/NightDreamUI.java:522]: Why is this modification necessary ?

File: src/com/firebirdberlin/nightdream/Settings.java [hunk 4]
```diff
@@ -95,6 +95,8 @@ public WeatherProvider getWeatherProvider() {
     public boolean muteRinger = false;
     public boolean persistentBatteryValueWhileCharging = true;
     public boolean restless_mode = true;
```
Comment (firebirdberlin) [inline src/com/firebirdberlin/nightdream/Settings.java:97]: How do you make sure, the setting `restless_mode` is transferred to the new settings ?

File: res/values/strings.xml [hunk 5]
```diff
@@ -273,7 +273,9 @@ OR
     <string name="remaining">rem.</string>
     <string name="repeat">repeat</string>
     <string name="reset_to_defaults">Reset to defaults</string>
+    <string name="preventBurnIn">Prevent the clock from burn in</string>
```
Comment (firebirdberlin) [inline res/values/strings.xml:276]:
```suggestion
    <string name="preventBurnIn">Screen burn protection</string>
```

File: res/values/strings.xml [hunk 6]
```diff
@@ -273,7 +273,9 @@ OR
...
     <string name="restless_mode">Move the clock around</string>
+    <string name="fade_clock">Fade clock</string>
```
Comment (firebirdberlin) [inline res/values/strings.xml:278]: For translations it is easier if all related strings are grouped together (the translation tool automatically sort alphabetically). What about naming the strings `screenProtection`, `screenProtectionFade` and `screenProtectionMove` ?

File: res/xml-v14/preferences_behaviour.xml [hunk 7]
```diff
@@ -15,8 +15,19 @@
     <SwitchPreferenceCompat
         app:iconSpaceReserved="false"
         app:defaultValue="true"
-        app:key="restlessMode"
-        app:title="@string/restless_mode" />
+        app:key="preventBurnIn"
```
Comment (firebirdberlin) [inline res/xml-v14/preferences_behaviour.xml:18]: We could get rid of this switch if we added a selection for 'None' to the ListPreference.

File: res/xml-v14/preferences_behaviour.xml [hunk 8]
```diff
@@ -15,8 +15,19 @@
...
+        app:title="@string/preventBurnIn"
+        app:isPreferenceVisible="true"
+        />
+
+    <ListPreference
+        app:iconSpaceReserved="false"
+        app:defaultValue="1"
+        app:entries="@array/preventBurnInMode"
+        app:entryValues="@array/preventBurnInModeValues"
+        app:key="preventBurnInMode"
```
Comment (firebirdberlin) [inline res/xml-v14/preferences_behaviour.xml:28]:
```suggestion
        app:key="screenProtectionMode"
```

File: src/com/firebirdberlin/nightdream/Settings.java [hunk 9]
```diff
@@ -351,6 +355,25 @@ public void reload() {
...
     }
 
+    boolean getPreventBurnIn() {
+        boolean on = settings.getBoolean("preventBurnIn", true);
+
+        if (!restless_mode){
+            on = false;
+            setPreventBurnIn(on);
+            setRestlessMode(true);
+        }
```
Comment (firebirdberlin) [inline src/com/firebirdberlin/nightdream/Settings.java:365]: It is not necessary to write restlessMode. We want to get rid of this setting. Instead I would do 

```suggestion
        if (settings.contains("restlessMode"){
            Editor edit = settings.edit()
            edit.setString("screenProtectionMode", 1)
            edit.remove("restlessMode");
            edit.apply;
        }
       return settings.getString("screenProtectionMode", 1); 
```

If you do so, you are able to delete all other occurrences of restless_mode in the code.

File: src/com/firebirdberlin/nightdream/Settings.java [hunk 10]
```diff
@@ -351,6 +353,14 @@ public void reload() {
         }
     }
 
+    int getScreenProtection() {
+        if (settings.contains("restlessMode") && (!settings.contains("restlessMode"))){
+            settings.edit().putString("screenProtection", "0").apply();
```
Comment (firebirdberlin) [inline src/com/firebirdberlin/nightdream/Settings.java:358]:
```suggestion
        if (settings.contains("restlessMode") && settings.getBoolean("restlessMode", true)){
            settings.edit().putString("screenProtection", "1").apply();
```

Comment (mastermichi) [reply src/com/firebirdberlin/nightdream/Settings.java]: fixed the bug.
But standard for "restlessMode" is true and standard for "screenProtection" is 1 (move clock around).
So we have to check for "restlessMode == false" and set "screenProtection" to 0.

Comment (firebirdberlin) [reply src/com/firebirdberlin/nightdream/Settings.java]: You are right ;)

**Comment by firebirdberlin** [comment]
merged

**Pull request closed without merging**